### PR TITLE
[fixes #841] ensure testEnvironment.{beforeEach,afterEach} are released

### LINKED ISSUE #841: [Memory Leak] beforeEach / afterEach appear to leak
the following code (from : [source](https://github.com/jquery/qunit/blob/825691fda42d89723ff6b1a7337c2e529124d479/src/test.js#L73-L75)) appears to leak `beforeEach` and `afterEach` on the original module. This results in per module leak of the ambient context.

```js
this.testEnvironment = extend( {}, this.module.testEnvironment );
delete this.testEnvironment.beforeEach;
delete this.testEnvironment.afterEach;
```

Obviously the following, prevents the leaks:

```js
var foo;

module('test', {
  beforeEach() {
    var foo = new Foo();
  },

  afterEach() {
    foo = null
  }
});
```

but it is tedious and error prone. A much better solution, for those using modules (AMD/CJS/etc), is merely to release the entirely module. Unfortunately, QUnit retaining `beforeEach` and `afterEach` per module, prevents a less error prone and tedious solution.

cc @cibernox

### PR DESCRIPTION
**Pull request by stefanpenner**
@cibernox & @stefanpenner

### CODE REVIEW

**Comment by jzaefferer** [comment]
Thank you for the contribution. Can you please sign our CLA? http://contribute.jquery.org/CLA/

**Comment by stefanpenner** [comment]
> Thank you for the contribution. Can you please sign our CLA? http://contribute.jquery.org/CLA/

Send it to legal, lets see what they have to say.

@jzaefferer can you provide feedback now, I can atleast iterate to make sure this lands quickly. This leak is causing grief on many large apps, so we should try to get this in as soon as possible.

File: src/test.js [hunk 1]
```diff
@@ -70,9 +70,11 @@ Test.prototype = {
 
 		config.current = this;
 
+		if (this.module.testEnvironment) {
```
Comment (jzaefferer) [inline src/test.js]: Minor style issue: Missing spaces inside `if`. Not sure why jscs didn't catch this.

**Comment by jzaefferer** [comment]
The actual change looks good. I can fix the style issue when landing this, if you don't get around to it.

**Comment by stefanpenner** [comment]
@jzaefferer style fixed, I'll push for +1 on the CLA quickly.

Are we looking at a quick turn-around on release once merged?

**Comment by leobalter** [comment]
I was looking at it, but then I had to stop for some scheduled meetings. It took me a while because I went to the original sources to take a deep look.

This fix seems good to me.

**Comment by leobalter** [comment]
It also seems ok to release QUnit 1.19.0 after this.

**Comment by leobalter** [comment]
I'm ready to merge jquery/qunit#842 I just need to confirm if you signed the CLA.

Merging it, I'll release QUnit 1.19, probably on Monday morning

**Comment by stefanpenner** [comment]
Ya, still pending legal :+1:.

**Comment by stefanpenner** [comment]
sorry for the delay folks, still pending legal :+1:

**Comment by leobalter** [comment]
ok, I'm holding the 1.19 release until we get the approval.

**Comment by stefanpenner** [comment]
Sorry about the delay, I have *just* gotten the legal :+1: and signed the CLA.

Apparently there was a small concern about some specific verbiage, if anyone is interested I can put them in contact with them. On a positive note, anyone at Yahoo! now has :+1: to contribute.

**Comment by leobalter** [comment]
Thanks, @stefanpenner! It's merged now.

**Comment by stefanpenner** [comment]
sorry for the delay

**Pull request closed without merging**